### ISSUE TITLE
go: command not found

### ISSUE THREAD

**Issue by iseolin76** [heroku/heroku-buildpack-go#467]
The go command does not work even though the go build pack is applied.
version: 1.16
Below is the corresponding log.

<img width="939" alt="스크린샷 2021-06-26 오전 12 12 08" src="https://user-images.githubusercontent.com/65021545/123445744-277a9600-d613-11eb-8210-0a97adc12285.png">

The following is my file directory

<img width="158" alt="스크린샷 2021-06-26 오전 12 17 43" src="https://user-images.githubusercontent.com/65021545/123446562-f8b0ef80-d613-11eb-846d-f590ac7232ca.png">

What did I miss?

**Comment by pemistahl** [comment]
I just stumbled over the same problem. The command in your `Procfile` is wrong. Instead of writing `web: go run main.go` you need to enter `web: /bin/your-app-name`. The app name is equivalent to the name of your git repository.

This is pretty badly documented, I must say. It took me quite some time to figure this out. The Heroku guys should improve their docs.

**Comment by mble** [comment]
Yes, we should make this clearer that the buildpack first builds your binaries and does not include a copy of the toolchain. We do explain this here https://devcenter.heroku.com/articles/go-support#runtime-behavior, but it is pretty out of the way.

**Comment by acamara2016** [comment]
I'm still experiencing this issue, the docs is not really clear on how to write your Procfile to run the go server.

**Comment by mble** [comment]
@acamara2016 Apologies for the difficulties - the Procfile should map a "process type" e.g. `web` to a command executing the binary post compile, e.g:

```Procfile
web: web-server -port "${PORT}"
```

You can see an example of this here: https://github.com/heroku/go-getting-started/blob/main/Procfile.

If you have multiple `main` packages (e.g. in a `cmd/server`, `cmd/worker` style setup) you would adjust your `go.mod` to include additional directives, such as:

```
// +heroku install ./cmd/server ./cmd/worker
```

Which would then be available through something like:

```Procfile
web: bin/server
worker: bin/worker
```

More details in https://github.com/heroku/heroku-buildpack-go?tab=readme-ov-file#go-module-specifics.

I'll look at adding a complete worked example to the docs tomorrow.
